kodi_addon_checker/schema_validation.py

```diff
@@ -0,0 +1,33 @@
+import xmlschema
+import logging
+
+from .report import Report
+from .record import PROBLEM, Record, INFORMATION
+
+xml_schema = "./xml_schema/"
+LOGGER = logging.getLogger(__name__)
+
+
+def schemas(report: Report, parsed_xml):
+
+    for extension in parsed_xml.findall("extension"):
+        extension_point = (extension.get("point")).split(".")[-1]
+        schema_path = xml_schema + extension_point + '.xsd'
+        validate(report, extension, schema_path)
+        parsed_xml.remove(extension)
+
+    root_schema = xml_schema + 'addon.xsd'
+    validate(report, parsed_xml, root_schema)
+
+
+def validate(report: Report, xml, schemapath):
+
+    schema = xmlschema.XMLSchema(schemapath)
+
+    try:
+        schema.validate(xml)
+        report.add(Record(INFORMATION, "Valid XML file found"))
```

Comment:
useless info, you expect that if no issue is logged it is valid

Reply:
If you want to report that it is valid then this is the wrong place as this might be called multiple times for the same addon.xml. It could also happen that it reports all fine and an error for the same addon which would be confusing.